### PR TITLE
Issue #265  Why doesn't Contact inherit from SyncSettingItem?

### PR DESCRIPTION
**Pull request by k-c13**
See #265 

Add extended SyncSettingItem class 	
Update TypeAdapter for using extended SyncSettingItem class 	
Update test

### CODE REVIEW

File: src/main/java/de/qabel/core/config/Contact.java [hunk 1]
```diff
@@ -178,6 +178,9 @@ protected Contact(String ownerKeyId) {
 	public int hashCode() {
 		final int prime = 31;
 		int result = 1;
```
Comment (aBothe) [inline src/main/java/de/qabel/core/config/Contact.java]: `= 1` and the line below should be merged.

**Comment by aBothe** [comment]
Looks fine though.

**Comment by cburkert** [comment]
Well, I did this already as part of my refactoring of the drop API (as I mentioned in my initial post on #265 but you were faster with your PR ;-)

I would name the SyncSettingItem fields explicitly in the TypeAdapter but use a SyncSettingItem-TypeAdapter for that. That makes it more flexible.
My PR may take some more days, so it should be no argument against merging this PR.

**Comment by jan-schreib** [comment]
We talked about the brackets in the meeting. Will be changed over time. @cburkert if you are ok with this pullreq I'll merge it.

**Comment by cburkert** [comment]
@k-c13 and I decided that I will integrate his approach in my PR #269 
So you this PR should not be merged. I'll close it if I updated my PR.

**Comment by jan-schreib** [comment]
@cburkert Ok.

**Comment by cburkert** [comment]
I adopted the flat serialization of inherited properties from this PR into my PR 60a97e9366373772efbd00368eaa7a4518ea2414
I'll close this one then.

**Pull request closed without merging**